packages/block-library/src/categories/edit.js

```diff
@@ -116,7 +116,11 @@ class CategoriesEdit extends Component {
 					{ __( 'Categories' ) }
 				</label>
 				<select id={ selectId } className="wp-block-categories__dropdown">
-					{ categories.map( ( category ) => this.renderCategoryDropdownItem( category, 0 ) ) }
+					{ categories.map( ( category ) => {
```

Comment:
This is the case where you can also use more functional approach and combine `filter` and `map`:
```js
{ categories.filter( ( { parent } => parent === 0 ).map( ( category ) => this.renderCategoryDropdownItem( category, 0 ) ) }
```

Still the same code, just coded differently and I haven't tested :)

Reply:
Hi @gziolo, I applied the refactor to use a functional code style. Thank you for your suggestion!